### PR TITLE
Redirect docs subdomain

### PR DESCRIPTION
**Pull request by chance19**
The config in the previous PR did not work, this PR will 
- Remove the config applied with #736
- Add redirect config to _redirects file in order to remain consistent with existing redirects

### CODE REVIEW

**Review by wallrj — APPROVED** [review]
/lgtm
/approve

File: _redirects [hunk 1]
```diff
@@ -35,3 +35,7 @@ https://cert-manager.io/docs/usage/kubectl-plugin/#status-certificate https://ce
 https://cert-manager.io/docs/usage/kubectl-plugin/#completion https://cert-manager.io/docs/usage/cmctl/#completion
 https://cert-manager.io/docs/usage/kubectl-plugin/#experimental https://cert-manager.io/docs/usage/cmctl/#experimental
 https://cert-manager.io/docs/usage/kubectl-plugin/#certificatesigningrequest https://cert-manager.io/docs/usage/cmctl/#certificatesigningrequest
+
+# docs.cert-manager.io was previously a separately hosted service. The dns has since been redirect to the main site and the following redirect
+# rule will force people with historical links or browser to docs.cert-manager.io to be redirected to the /docs endpoint
+https://docs.cert-manager.io/* https://cert-manager.io/docs/
```
Comment (munnerz) [inline _redirects:41]: Will this retain the trailing path segments so they go to the correct page?

**Comment by munnerz** [comment]
Hm I think this has happened after I deleted the old readthedocs site - that site was serving redirects from old config to the new pages.

Given those aren't 1:1 either (some content was re-organised) we may want to consider porting across the old rules... I'll see if I can dig them out!

**Comment by munnerz** [comment]
Here is the full list: https://github.com/jetstack/cert-manager/blob/release-0.12/docs/external_redirects

**Comment by munnerz** [comment]
Which may also want integrating with this: https://github.com/jetstack/cert-manager/blob/release-0.12/docs/redirects for completeness..

File: _redirects (hunk 1 above)
Comment (mattbates) [reply _redirects:41]: It won't - we need to add `the :splat` to the redirected URL.

Comment (munnerz) [reply _redirects:41]: The issue is also that we re-mapped the URLs when moving too I think, see https://github.com/jetstack/cert-manager/blob/release-0.12/docs/redirects

Comment (munnerz) [reply _redirects:41]: We could set up a small webservice that just serves 301 redirects to fix this up perhaps?

Comment (maelvls) [reply _redirects:41]: Good idea. 

Should we map all the URLs found on Google? E.g.,

https://docs.cert-manager.io/en/release-0.9/reference/issuers.html maps to https://cert-manager.io/docs/concepts/issuer/?

I did a quick google search, here are the existing links:

```
https://docs.cert-manager.io/en/release-0.11
https://docs.cert-manager.io/en/release-0.8/
https://docs.cert-manager.io/en/release-0.10/
https://docs.cert-manager.io/en/release-0.9/
https://docs.cert-manager.io/en/release-0.7/
https://docs.cert-manager.io/_/downloads/en/release-0.7/pdf/
https://docs.cert-manager.io/_/downloads/en/release-0.9/pdf/
https://docs.cert-manager.io/en/release-0.9/reference/issuers.html
https://docs.cert-manager.io/en/release-0.11/reference/challenges.html
https://docs.cert-manager.io/_/downloads/en/release-0.10/pdf/
https://docs.cert-manager.io/_/downloads/en/release-0.10/pdf/
https://docs.cert-manager.io/en/release-0.11/reference/certificates.html
https://docs.cert-manager.io/en/release-0.8/reference/orders.html
https://docs.cert-manager.io/en/release-0.11/reference/certificaterequests.html
https://docs.cert-manager.io/en/release-0.9/reference/orders.html
https://docs.cert-manager.io/en/release-0.9/getting-started/webhook.html
https://docs.cert-manager.io/en/release-0.8/reference/clusterissuers.html
https://docs.cert-manager.io/en/release-0.11/reference/clusterissuers.html
https://docs.cert-manager.io/en/release-0.11/getting-started/webhook.html
https://docs.cert-manager.io/en/release-0.7/getting-started/troubleshooting.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/index.html
https://docs.cert-manager.io/en/release-0.7/getting-started/install.html
https://docs.cert-manager.io/en/release-0.11/tasks/issuers/index.html
https://docs.cert-manager.io/en/release-0.10/tasks/issuers/index.html
https://docs.cert-manager.io/en/release-0.9/devel/dns01-providers.html
https://docs.cert-manager.io/en/release-0.11/reference/cainjector.html
https://docs.cert-manager.io/en/release-0.11/reference/api-docs/index.html
https://docs.cert-manager.io/en/release-0.9/reference/api-docs/index.html
https://docs.cert-manager.io/en/release-0.9/tasks/issuers/setup-ca.html
https://docs.cert-manager.io/en/release-0.11/tasks/issuers/setup-ca.html
https://docs.cert-manager.io/en/release-0.10/getting-started/install/kubernetes.html
https://docs.cert-manager.io/en/release-0.7/reference/api-docs/index.html
https://docs.cert-manager.io/en/release-0.11/getting-started/install/kubernetes.html
https://docs.cert-manager.io/en/release-0.8/reference/api-docs/index.html
https://docs.cert-manager.io/en/release-0.9/getting-started/index.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/issuing-certificates.html
https://docs.cert-manager.io/en/release-0.7/tasks/issuing-certificates/index.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-ca.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-vault.html
https://docs.cert-manager.io/en/release-0.7/tasks/issuers/setup-vault.html
https://docs.cert-manager.io/en/release-0.11/tasks/issuers/setup-vault.html
https://docs.cert-manager.io/en/release-0.10/devel/develop-with-minikube.html
https://docs.cert-manager.io/en/release-0.11/devel/develop-with-minikube.html
https://docs.cert-manager.io/en/release-0.7/tasks/issuers/setup-ca.html
https://docs.cert-manager.io/en/release-0.7/devel/develop-with-minikube.html
https://docs.cert-manager.io/en/release-0.11/tutorials/acme/index.html
https://docs.cert-manager.io/en/release-0.8/devel/develop-with-minikube.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/configuring-http01.html
https://docs.cert-manager.io/en/release-0.11/tasks/issuers/setup-acme/index.html
https://docs.cert-manager.io/en/release-0.11/tasks/uninstall/index.html
https://docs.cert-manager.io/en/release-0.11/tasks/backup-restore-crds.html
https://docs.cert-manager.io/en/release-0.7/tasks/issuing-certificates/ingress-shim.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/configuring-dns01/google.html
https://docs.cert-manager.io/en/release-0.7/tasks/issuers/setup-selfsigned.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuing-certificates/ingress-shim.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-acme/index.html
https://docs.cert-manager.io/en/release-0.9/tasks/acme/configuring-dns01/google.html
https://docs.cert-manager.io/en/release-0.10/tasks/acme/configuring-dns01/azuredns.html
https://docs.cert-manager.io/en/release-0.10/tasks/issuing-certificates/ingress-shim.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-selfsigned.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/configuring-dns01/azuredns.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/configuring-dns01/index.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/configuring-dns01/route53.html
https://docs.cert-manager.io/en/release-0.11/getting-started/install/index.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-acme/dns01/google.html
https://docs.cert-manager.io/en/release-0.10/tasks/issuers/setup-acme/dns01/google.html
https://docs.cert-manager.io/en/release-0.9/tasks/acme/configuring-dns01/cloudflare.html
https://docs.cert-manager.io/en/release-0.7/tasks/acme/debugging-failing-orders.html
https://docs.cert-manager.io/en/release-0.9/tasks/acme/configuring-dns01/route53.html
https://docs.cert-manager.io/en/release-0.9/tasks/acme/configuring-dns01/digitalocean.html
https://docs.cert-manager.io/en/release-0.11/tasks/issuers/setup-acme/dns01/azuredns.html
https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-acme/dns01/route53.html
https://docs.cert-manager.io/en/release-0.9/tasks/issuers/setup-acme/dns01/akamai.html
```

<details>
<summary>Reproduce</summary>

1. get a `key` at https://developers.google.com/custom-search/v1/overview
2. Google a Search Engine ID (`cx`) at https://programmablesearchengine.google.com/controlpanel/all
3. Run:

```sh
for i in $(seq 0 10 100); do curl -s "https://customsearch.googleapis.com/customsearch/v1?key=$(lpass show -p google-search)&cx=$(lpass show -u google-search)&start=$i&q=site%3Adocs.cert-manager.io" | jq -r '.items[].link'; done
```

</details>

**Pull request merged**